File: 2025/day01/2.c
#include <stdio.h>

int main(){
	int a=50, ans=0;
	char s[20];

	while (fgets(s, sizeof s, stdin)) {
		int x;
		char c;
		sscanf(s, " %c%d", &c, &x);

		if (c=='L') {
			while (x--) {
				a--;
				if (a==-1) a=99;
				if (a==0) ans++;
			}
		} else {
			while (x--) {
				a++;
				if (a==100) {
					a=0;
					ans++;
				}
			}
		}
	}

	printf("%d", ans);
}

File: 2025/day01/2.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let reader = stdin.lock();

    let mut a: i32 = 50;
    let mut ans: i32 = 0;

    for line in reader.lines() {
        let text = line.unwrap();
        let c = text.chars().nth(0).unwrap();
        let n: i32 = text[1..].trim().parse().unwrap();

        match c {
            'L' => {
                for _ in 0..n {
                    a -= 1;
                    if a < 0 {
                        a = 99;
                    }
                    if a == 0 {
                        ans += 1;
                    }
                }
            }
            'R' => {
                for _ in 0..n {
                    a += 1;
                    if a >= 100 {
                        a = 0;
                    }
                    if a == 0 {
                        ans += 1;
                    }
                }
            }
            _ => panic!("Invalid input"),
        }
    }

    println!("{}", ans);
}
